### ISSUE TITLE
Progress formatter creating larger cells than expected

### ISSUE THREAD

**Issue by phallett** [olifolkerd/tabulator#2136]
In using a progress formatter, the height of a cell becomes very large in showing the data (I have included a screenshot.  Hard to explain, but instead of a 'normal' sized cell, it elongates in height, making the table hard to read.

I am populating the table with an Ajax call that returns a Json string of user details.  It populates the data perfectly, but renders the cell in the way I have described.  I would expect the cell to render with a progress bar indicating a color and number based on the value.

Table code:
` 
    $.ajax({
                url : "/getuserotherdeptdetails", //"{{url_for('checkalloc')}}", 
                data: {userid:userid},
                type: 'GET',
                contentType: "application/json;",
                success: function(data){

                    var obj = $.parseJSON(data);
                    if(data.length>2){
                        var table = new Tabulator("#user-non-colleague-table", {
                            layout:"fitData",
                            data:obj,           //load row data from array
                            tooltips:true,            //show tool tips on cells
                            addRowPos:"bottom",          //when adding a new row, add it to the top of the table
                            history:true,             //allow undo and redo actions on the table
                            pagination:"local",       //paginate the data
                            paginationSize:15,         //allow 7 rows per page of data
                            movableColumns:false,      //allow column order to be changed
                            resizableRows:false,       //allow row order to be changed,
                            columns:[
                            {title:"Name", field:"Name", frozen:true, headerFilter:"input"},
                            {title:"Week 1", field:"wk1", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 2", field:"wk2", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 3", field:"wk3", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 4", field:"wk4", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 5", field:"wk5", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 6", field:"wk6", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 7", field:"wk7", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 8", field:"wk8", formatter:"progress", formatterParams:paramLookup},
                            {title:"Week 9", field:"wk9", formatter:"progress", formatterParams:paramLookup},
    ],
    });     
    },
    error: function(error){
    console.log(error);
    }
    });
`

The paramLookup function is as :
`    
    var paramLookup = function(cell, formatterParams, onRendered){

        var val = cell.getValue();   
        baseval = 37;
        lowval = baseval * .7;
        highval =  baseval * 1.3;
        var ret = "yellow"

        if (val <= lowval && val >= 1){
            ret = "yellow"       
        } else if (val > lowval && val < highval){
            ret = "green"
        } else if (val >= highval){
            ret = "red"
        } 
        return {color:ret,legend:true, min:0, max:48};
    };
`

I'm using this on iOS 10.14.5 Mojave using Tabulator 4.2.7 on Google Chrome Version 74.0.3729.169,  Safari Version 12.1.1 (14607.2.6.1.1) and Firefox version 66.0.5.

<img width="1075" alt="Screenshot 2019-06-10 at 11 40 14" src="https://user-images.githubusercontent.com/676839/59192240-b8d85600-8b79-11e9-90d4-cd37108a8954.png">

If you have a look at the screenshot, I'd expect the cell with the progress formatter to be the same size as the one in the table at the top, formatted to the correct color.  It creates the color perfectly, but just enlarges the cell and I can't see how or why?

I am unsure if I am doing something wrong, but to me it looks like I have followed the correct instructions.  It's a wonderful bit of kit tabulator, so I am desperate to get this bit working.

An indication of the cell sizes are indicated here>. the width is ok, it's just the height.  The following screenshot is almost my entire screen.

<img width="1408" alt="Screenshot 2019-06-10 at 12 21 48" src="https://user-images.githubusercontent.com/676839/59192468-77947600-8b7a-11e9-967f-bb1189532241.png">

**Comment by phallett** [comment]
Canning down on the data, I can get it to work, but it's precisely the same code, with less data, so I am completely confused?

<script type="text/javascript">

$(document).ready(function(element) {
  fetchProjectMembers();
    function paramLookup (cell){

        var val = cell.getValue();   
        baseval = 37;
        lowval = baseval * .7;
        highval =  baseval * 1.3;

        if (val <= lowval){
            return {color:"yellow",legend:true, min:0, max:48};    
        } else if (val > lowval && val < highval){
            return {color:"green",legend:true, min:0, max:48};
        } else if (val >= highval){
            return {color:"red",legend:true, min:0, max:48};
        }    
        
    };

    var colorFormatter = function(cell, formatterParams, onRendered){

        var val = cell.getValue();   
        baseval = 37;
        lowval = baseval * .7;
        highval =  baseval * 1.3;

        if (val <= lowval){
            cell.getElement().style.backgroundColor="yellow";
            
        } else if (val > lowval && val < highval){
            cell.getElement().style.backgroundColor="green";
            cell.getElement().style.color="white";
        } else if (val >= highval){
            cell.getElement().style.backgroundColor="red";
            cell.getElement().style.color="white";
        }

        return val;
    };


  function fetchProjectMembers(){ 

    tabledata = [{"Name": "Monkey 1", "wk1": 102.0, "wk2": 49.0, "wk3": 33.3, "wk4": 33.3, "wk5": 33.3, "wk6": 37.0, "wk7": 37.0, "wk8": 37.0, "wk9": 37.0, "wk10": 37.0}, {"Name": "Monkey 2", "wk1": 0.0, "wk2": 37.0, "wk3": 38.85, "wk4": 38.85, "wk5": 75.85, "wk6": 38.85, "wk7": 38.85, "wk8": 38.85, "wk9": 75.85, "wk10": 55.85}, {"Name": "Monkey 3", "wk1": 37.0, "wk2": 55.0, "wk3": 37.0, "wk4": 37.0, "wk5": 37.0, "wk6": 54.0, "wk7": 37.0, "wk8": 37.0, "wk9": 37.0, "wk10": 37.0}]

    var table = new Tabulator("#user-table", {
      layout:"fitData",
      data:tabledata,           //load row data from array
      //autoColumns:true,    //fit columns to width of table
      //responsiveLayout:"hide",  //hide columns that dont fit on the table
      tooltips:true,            //show tool tips on cells
      addRowPos:"bottom",          //when adding a new row, add it to the top of the table
      history:true,             //allow undo and redo actions on the table
      pagination:"local",       //paginate the data
      paginationSize:15,         //allow 7 rows per page of data
      movableColumns:false,      //allow column order to be changed
      resizableRows:false,       //allow row order to be changed,
      columns:[
        {title:"Name", field:"Name", frozen:true, headerFilter:"input"},
        {title:"Week 1", field:"wk1", bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 2", field:"wk2",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 3", field:"wk3",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 4", field:"wk4",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 5", field:"wk5",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 6", field:"wk6",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 7", field:"wk7",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 8", field:"wk8",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 9", field:"wk9",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
        {title:"Week 10", field:"wk10",bottomCalc:"sum", bottomCalcParams:{precision:2,}, bottomCalcFormatter: colorFormatter},
      ],
    });
      var table = new Tabulator("#user-table-2", {
      layout:"fitData",
      data:tabledata,           //load row data from array
      //autoColumns:true,    //fit columns to width of table
      //responsiveLayout:"hide",  //hide columns that dont fit on the table
      tooltips:true,            //show tool tips on cells
      addRowPos:"bottom",          //when adding a new row, add it to the top of the table
      history:true,             //allow undo and redo actions on the table
      pagination:"local",       //paginate the data
      paginationSize:15,         //allow 7 rows per page of data
      movableColumns:false,      //allow column order to be changed
      resizableRows:false,       //allow row order to be changed,
      columns:[
        {title:"Name", field:"Name", frozen:true, headerFilter:"input"},
        {title:"Week 1", field:"wk1", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 2", field:"wk2", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 3", field:"wk3", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 4", field:"wk4", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 5", field:"wk5", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 6", field:"wk6", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 7", field:"wk7", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 8", field:"wk8", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 9", field:"wk9", formatter:"progress", formatterParams:paramLookup},
        {title:"Week 10", field:"wk10", formatter:"progress", formatterParams:paramLookup},
      ],
    });
  };
});
                           
</script>

<div class="card ">
  <div class="card-body">
    <h5 class="card-header">staff </h5>
    <div id="user-table"></div><br>
      <h5 class="card-header">Other staff </h5>
    <div id="user-table-2"></div><br>
  </div>
</div>

**Comment by tentus** [comment]
Your code is a bit hard to follow. Please use this fiddle as a basis to get a demonstration of your error going:

https://jsfiddle.net/row8mtkg/

**Comment by phallett** [comment]
Thanks, but I couldn't get fiddle working for some reason.

The problem is, fiddle is proving it works, but the code is the same, without hard coding the data.

**Comment by phallett** [comment]
I have solved this, apologies.  It was due to malformed HTML tags

https://github.com/olifolkerd/tabulator/issues/2076#issuecomment-500760090
